client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/coverage/methodreplacement/thirdpartyclasses/OkUrlFactoryClassReplacement.java

```diff
@@ -75,6 +76,10 @@ public static HttpURLConnection open(Object caller, URL url, Proxy proxy) {
     }
 
     private static URL getReplacedURL(URL url){
+        if (MethodReplacementPreserveSemantics.shouldPreserveSemantics) {
```

Comment:
noticed one issue... isn't this very similar to what done in `URLClassReplacement`? 
should avoid code duplication. 
should be refactored in a utility function. 

also, here it seems it is missing the handling of query parameters, ie:

```
if (caller.getQuery() != null)
                url = url + "?" + caller.getQuery();
```

anyway, such refactoring can be done in a different branch. please do

Reply:
hi @seran you marked this as resolved? how did you resolve it? is it done in a different branch? if not (yet), should write a TODO comment here in the code